src/lib/index.js

```diff
@@ -1008,6 +1008,60 @@ lib.templateString = function(string, obj) {
     });
 };
 
+/*
+ * Substitute values from an object into a string and optionally formats them using d3-format,
+ * or fallback to associated labels.
+ *
+ * Examples:
+ *  Lib.templateString('name: %{trace}', {trace: 'asdf'}) --> 'name: asdf'
+ *  Lib.templateString('name: %{trace[0].name}', {trace: [{name: 'asdf'}]}) --> 'name: asdf'
+ *  Lib.templateString('price: %{y:$.2f}', {y: 1}) --> 'price: $1.00'
+ *
+ * @param {string}  input string containing %{...:...} template strings
+ * @param {obj}     data object containing fallback text when no formatting is specified, ex.: {yLabel: 'formattedYValue'}
+ * @param {obj}     data objects containing substitution values
+ *
+ * @return {string} templated string
+ */
+
+lib.hovertemplateString = function(string, labels) {
+    var args = arguments;
+    // Not all that useful, but cache nestedProperty instantiation
+    // just in case it speeds things up *slightly*:
+    var getterCache = {};
+
+    return string.replace(TEMPLATE_STRING_REGEX, function(match, key, format) {
+        var obj, value, i;
+        for(i = 2; i < args.length; i++) {
+            obj = args[i];
+            if(obj.hasOwnProperty(key)) {
+                value = obj[key];
+                break;
+            }
+
+            if(!SIMPLE_PROPERTY_REGEX.test(key)) {
+                // getterCache[key] = getterCache[key] || lib.nestedProperty(obj, key).get;
+                // value = getterCache[key]();
+                value = getterCache[key] || lib.nestedProperty(obj, key).get();
+                if(value) getterCache[key] = value;
+            }
+            if(value !== undefined) break;
+        }
+
+        if(value === undefined) {
+            lib.warn('Variable \'' + key + '\' in hovertemplate could not be found!');
```

Comment:
We should probably add a limit to how many of these `lib.warn` get printed in the console. As these happen during mousemove, we wouldn't want to flood the console with identical warning messages.

Reply:
Would something like throttling be satisfactory? I am pondering what to do :thinking:

Reply:
I would just define a variable e.g. `numberOfHoverTemplateWarnings` in the outer scope, increment it here, and skip this `lib.warn` call here if greater than e.g. `10`.

Reply:
Should we reset `numberOfHoverTemplateWarnings` to 0 each time `hovertemplate` changes? We probably don't want this function to go quiet forever :thinking:

Reply:
I wouldn't worry about bringing the messages back, but do let the user know what happened - ie after the last displayed message add another like
`Lib.warn('Too many hovertemplate warnings - additional warnings will be suppressed')`

Reply:
> Should we reset `numberOfHoverTemplateWarnings` to 0 each time `hovertemplate` changes?

That would be nice, but not necessary for now in my opinion. Feel free to try to implement it if you're up for a challenge though.

Reply:
Done in 3068d7d

Reply:
BTW, I went with Alex's suggestion for now